yb-voyager/src/query/queryparser/helpers_struct.go

```diff
@@ -45,6 +45,22 @@ func IsMviewObject(parseTree *pg_query.ParseResult) bool {
 	return isCreateAsStmt && createAsNode.CreateTableAsStmt.Objtype == pg_query.ObjectType_OBJECT_MATVIEW
 }
 
+func getDefineStmtNode(parseTree *pg_query.ParseResult) (*pg_query.DefineStmt, bool) {
+	node, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_DefineStmt)
+	return node.DefineStmt, ok
+}
+
+func IsCollationObject(parseTree *pg_query.ParseResult) bool {
+	collation, ok := getDefineStmtNode(parseTree)
+	/*
+		stmts:{stmt:{define_stmt:{kind:OBJECT_COLLATION  defnames:{string:{sval:"ignore_accents"}}  definition:{def_elem:{defname:"provider"
+		arg:{type_name:{names:{string:{sval:"icu"}}  typemod:-1  location:48}}  defaction:DEFELEM_UNSPEC  location:37}}  definition:{def_elem:{defname:"locale"
+		arg:{string:{sval:"und-u-ks-level1-kc-true"}}  defaction:DEFELEM_UNSPEC  location:55}}  definition:{def_elem:{defname:"deterministic"
+		arg:{string:{sval:"false"}}  defaction:DEFELEM_UNSPEC  location:91}}}}  stmt_len:113}
+	*/
+	return ok && collation.Kind == pg_query.ObjectType_OBJECT_COLLATION
```

Comment:
if we are dealing with structs, i dont think the need of adding parse tree here. For readability we can check open the struct and check the fields present.

Reply:
yeah, just keeping it for reference.